v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandlerTest.java

```diff
@@ -15,1303 +15,1137 @@
+
+    byte[] expectedBytes = new byte[] {1, 2, 3, 4, 5};
+    JSONObject valuesJson = new JSONObject();
+    valuesJson.put(columnName, expectedBytes);
+
+    SpannerColumnDefinition spannerColDef = new SpannerColumnDefinition(columnName, spannerType);
+    SourceColumnDefinition sourceColDef = new SourceColumnDefinition(columnName, sourceColType);
+
+    Object result = getColumnValueByType(spannerColDef, sourceColDef, valuesJson, "UTC");
+
+    assertNotNull(result);
+    assertTrue(result instanceof PreparedStatementValueObject);
+
+    Object actualValue = ((PreparedStatementValueObject<?>) result).value();
+    assertArrayEquals(expectedBytes, (byte[]) actualValue);
   }
 
   @Test
-  public void testConvertToCassandraDate_timezoneOffsetImpact() {
-    String dateString = "2024-12-16T14:30:00+01:00";
-    LocalDate result = CassandraTypeHandler.convertToCassandraDate(dateString);
+  public void testCastToExpectedTypeForVariousTypes() throws UnknownHostException {
+    assertEquals("Test String", castToExpectedType("text", "Test String"));
+    assertEquals(123L, castToExpectedType("bigint", "123"));
+    assertEquals(true, castToExpectedType("boolean", "true"));
+    assertEquals(
+        new BigDecimal("123.456"),
+        castToExpectedType("decimal", new BigDecimal("123.456").toString()));
+    assertEquals(123.456, castToExpectedType("double", "123.456"));
+    assertEquals(123.45f, ((Double) castToExpectedType("float", "123.45")).floatValue(), 0.00001);
+    assertEquals(InetAddress.getByName("127.0.0.1"), castToExpectedType("inet", "127.0.0.1"));
+    assertEquals(123, castToExpectedType("int", "123"));
+    assertEquals((short) 123, castToExpectedType("smallint", "123"));
+    assertEquals(
+        UUID.fromString("123e4567-e89b-12d3-a456-426614174000"),
+        castToExpectedType("uuid", "123e4567-e89b-12d3-a456-426614174000"));
+    assertEquals((byte) 100, castToExpectedType("tinyint", "100"));
     assertEquals(
-        "The parsed LocalDate should be '2024-12-16' regardless of timezone.",
-        LocalDate.of(2024, 12, 16),
-        result);
+        new BigInteger("123456789123456789123456789"),
+        castToExpectedType("varint", "123456789123456789123456789"));
+    String timeString = "14:30:45";
+    // Parse the time
+    LocalTime localTime = LocalTime.parse(timeString, DateTimeFormatter.ISO_TIME);
+    Object localTime1 = castToExpectedType("time", "14:30:45");
+    assertTrue(localTime1 instanceof LocalTime);
+    assertEquals(
+        Duration.ofHours(5), castToExpectedType("duration", Duration.ofHours(5).toString()));
   }
 
   @Test
-  public void testConvertToCassandraDate_validDateWithOffset() {
-    String dateString = "2024-12-16T14:30:00+01:00";
-    LocalDate result = CassandraTypeHandler.convertToCassandraDate(dateString);
-    assertEquals("The parsed LocalDate should be '2024-12-16'", LocalDate.of(2024, 12, 16), result);
+  public void testCastToExpectedTypeForJSONArrayToSet() {
```

Comment:
this test is for JSONArray as string rename the test and add a test for JSONArray(without toString) to set